src/ucp/rma/put_offload.c

```diff
@@ -14,8 +14,68 @@
 #include <ucp/core/ucp_request.inl>
 #include <ucp/dt/datatype_iter.inl>
 #include <ucp/proto/proto_multi.inl>
+#include <ucp/proto/proto_single.inl>
 
 
+static ucs_status_t ucp_proto_put_offload_short_progress(uct_pending_req_t *self)
+{
+    ucp_request_t                   *req = ucs_container_of(self, ucp_request_t,
+                                                            send.uct);
+    const ucp_proto_single_priv_t *spriv = req->send.proto_config->priv;
+    ucp_ep_t                         *ep = req->send.ep;
+    ucs_status_t status;
+    uct_rkey_t tl_rkey;
+
+    tl_rkey = ucp_rma_request_get_tl_rkey(req, spriv->super.rkey_index);
+    status  = uct_ep_put_short(ep->uct_eps[spriv->super.lane],
+                               req->send.dt_iter.type.contig.buffer,
+                               req->send.dt_iter.length,
+                               req->send.rma.remote_addr, tl_rkey);
+    if (ucs_unlikely(status == UCS_ERR_NO_RESOURCE)) {
+        req->send.lane = spriv->super.lane; /* for pending add */
+        return status;
+    }
+
+    /* UCS_INPROGRESS is not expected */
+    ucs_assert((status == UCS_OK) || UCS_STATUS_IS_ERR(status));
```

Comment:
According to uct_ep_put_short documentation INprogress can be returned. We cannot just make the assumption that it cannot be returned. Looks very tacky.

Reply:
@shamisp can you pls point to this documentation? 
In UCT API, only functions which have uct_completion_t argument can return UCS_INPROGRESS (such as flush or zcopy)